File: fix_findbyid.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
path = '/app/dist/repositories/objectRepository.js'
with open(path, 'r', encoding='utf-8', errors='replace') as f:
    content = f.read()

old = (
    'let query = `\n'
    '      SELECT o.*, \n'
    '             t.name as client_name,\n'
    '             g.name as group_name\n'
    '      FROM objects o\n'
    "      LEFT JOIN tiers t ON o.client_id = t.id AND t.type = 'CLIENT'\n"
    '      LEFT JOIN groups g ON o.group_id = g.id\n'
    '      WHERE o.id = $1\n'
    '    `;'
)

new = (
    'let query = `\n'
    '      SELECT o.*, \n'
    '             t.name as client_name,\n'
    '             g.name as group_name,\n'
    '             lp.latitude as location_lat,\n'
    '             lp.longitude as location_lng,\n'
    '             lp.time as last_updated,\n'
    '             COALESCE(o.address, lp.address) as address\n'
    '      FROM objects o\n'
    "      LEFT JOIN tiers t ON o.client_id = t.id AND t.type = 'CLIENT'\n"
    '      LEFT JOIN groups g ON o.group_id = g.id\n'
    '      LEFT JOIN LATERAL (\n'
    '        SELECT latitude, longitude, time, address\n'
    '        FROM positions\n'
    '        WHERE object_id = o.id\n'
    '        ORDER BY time DESC\n'
    '        LIMIT 1\n'
    '      ) lp ON true\n'
    '      WHERE o.id = $1\n'
    '    `;'
)

if old in content:
    content = content.replace(old, new, 1)
    with open(path, 'w', encoding='utf-8') as f:
        f.write(content)
    print('OK: findByIdWithJoins patched')
else:
    print('SKIP: pattern not found')
    # Debug: show surrounding area
    idx = content.find('findByIdWithJoins')
    if idx >= 0:
        print('--- context ---')
        print(repr(content[idx:idx+500]))

File: fix_findbyid.js
const fs = require('fs');
const path = '/app/dist/repositories/objectRepository.js';
let content = fs.readFileSync(path, 'utf8');

// Find the findByIdWithJoins function and locate the query
const marker = 'findByIdWithJoins(id, tenantId, clientId)';
const idx = content.indexOf(marker);
if (idx < 0) { console.log('MARKER NOT FOUND'); process.exit(1); }

// Find the backtick query block
const qStart = content.indexOf('let query = `', idx);
const qEnd = content.indexOf('`;', qStart) + 2;
const orig = content.slice(qStart, qEnd);
console.log('Original query:');
console.log(orig);
console.log('---');

const newQuery = `let query = \`
      SELECT o.*,
             t.name as client_name,
             g.name as group_name,
             lp.latitude as location_lat,
             lp.longitude as location_lng,
             lp.time as last_updated,
             COALESCE(o.address, lp.address) as address
      FROM objects o
      LEFT JOIN tiers t ON o.client_id = t.id AND t.type = 'CLIENT'
      LEFT JOIN groups g ON o.group_id = g.id
      LEFT JOIN LATERAL (
        SELECT latitude, longitude, time, address
        FROM positions
        WHERE object_id = o.id
        ORDER BY time DESC
        LIMIT 1
      ) lp ON true
      WHERE o.id = $1
    \``;

content = content.slice(0, qStart) + newQuery + content.slice(qEnd);
fs.writeFileSync(path, content, 'utf8');
console.log('OK: findByIdWithJoins patched');
